docs/spec/nmt-wrapper.md

```diff
@@ -0,0 +1,83 @@
+# Namespaced Merkle Tree Wrapper
+## Abstract
+In Celestia, block transactions are grouped into identical-size shares and arranged in a `k` by `k` matrix, called [original data square](https://github.com/celestiaorg/celestia-app/blob/specs-staging/specs/src/specs/data_structures.md#arranging-available-data-into-shares).
+The rows and columns of the original data square are then extended using a 2D Reed-Solomon coding scheme.
+The [extended version](https://github.com/celestiaorg/celestia-app/blob/specs-staging/specs/src/specs/data_structures.md#2d-reed-solomon-encoding-scheme) is called extended data square, that is a `2k` by `2k` square consisting of `4` quadrants namely, `Q0`, `Q1`, `Q2`, and `Q3`.
+Figure 1 provides an illustration of a sample data square and its extended version.
+`Q0` corresponds to the original data square.
+`Q1` and `Q2` represent the horizontal and vertical extensions of `Q0`, respectively. 
+`Q3` is the extension of `Q2` in the horizontal direction.
```

Comment:
Should we talk about if there is a reason Q2 is extending horizontally to form Q3 vs Q1 being extended vertically to make Q3?

Reply:
That is a good question! The mechanics of erasure coding are not within the scope of the NMT wrapper, and this logic is implemented in the caller application.  I recommend elaborating on this topic in the following document, which explains the erasure coding of the original data square: https://github.com/celestiaorg/celestia-app/blob/c09843d07d4c3842753138de96b304b4866e8f5d/specs/src/specs/data_structures.md#2d-reed-solomon-encoding-scheme.
cc: @evan-forbes @rootulp what do you think?

Reply:
> Should we talk about if there is a reason Q2 is extending horizontally to form Q3 vs Q1 being extended vertically to make Q3?

One more point: The description of the extended data square construction is provided in the abstract for additional context, if you think this causes confusion, I can either revise or remove that part.

Reply:
> I recommend elaborating on this topic in the following document, which explains the erasure coding of the original data square:

100% agree

Reply:
+1 agreed, seems out of scope for NMT wrapper but in-scope for specs. 

> Should we talk about if there is a reason Q2 is extending horizontally to form Q3 vs Q1 being extended vertically to make Q3?

My impression from @adlerjohn 's [whiteboard series w/ NEAR](https://youtu.be/jM-om3AqH94?t=2970) is that it doesn't matter if Q3 is extended from Q1 or Q2.

Reply:
sounds good, I thought it might apply somewhere else.

if you've captured the follow up then you can resolve this 👍

Reply:
Created https://github.com/celestiaorg/celestia-app/issues/1454